core/capabilities/ccip/oraclecreator/bootstrap.go

```diff
@@ -104,5 +133,241 @@ func (i *bootstrapOracleCreator) Create(_ uint32, config cctypes.OCR3ConfigWithM
 	if err != nil {
 		return nil, err
 	}
-	return bootstrapper, nil
+
+	bootstrapperWithCustomClose := newWrappedOracle(
+		bootstrapper,
+		[]io.Closer{pgd, rmnHomeReader},
+	)
+
+	return bootstrapperWithCustomClose, nil
+}
+
+func (i *bootstrapOracleCreator) getRmnHomeReader(ctx context.Context, config cctypes.OCR3ConfigWithMeta) (ccipreaderpkg.RMNHome, error) {
+	rmnHomeBoundContract := types.BoundContract{
+		Address: "0x" + hex.EncodeToString(config.Config.RmnHomeAddress),
+		Name:    consts.ContractNameRMNHome,
+	}
+
+	if err1 := i.contractReader.Bind(ctx, []types.BoundContract{rmnHomeBoundContract}); err1 != nil {
+		return nil, fmt.Errorf("failed to bind RMNHome contract: %w", err1)
+	}
+	rmnHomeReader := ccipreaderpkg.NewRMNHomePoller(
+		i.contractReader,
+		rmnHomeBoundContract,
+		i.lggr,
+		5*time.Second,
+	)
+	return rmnHomeReader, nil
+}
+
+// peerGroupDialer keeps watching for config changes and calls NewPeerGroup when needed.
+// Required for managing RMN related peer group connections.
+type peerGroupDialer struct {
+	lggr logger.Logger
+
+	peerGroupFactory rmn.PeerGroupFactory
+	rmnHomeReader    ccipreaderpkg.RMNHome
+
+	// common oracle config
+	bootstrapLocators  []commontypes.BootstrapperLocator
+	oraclePeerIDs      []ragep2ptypes.PeerID
+	commitConfigDigest [32]byte
+
+	activePeerGroups    []rmn.PeerGroup
+	activeConfigDigests []cciptypes.Bytes32
+
+	syncInterval time.Duration
+
+	mu        *sync.Mutex
+	syncCtxCf context.CancelFunc
+}
+
+func newPeerGroupDialer(
+	lggr logger.Logger,
+	peerGroupFactory rmn.PeerGroupFactory,
+	rmnHomeReader ccipreaderpkg.RMNHome,
+	bootstrapLocators []commontypes.BootstrapperLocator,
+	oraclePeerIDs []ragep2ptypes.PeerID,
+	commitConfigDigest [32]byte,
+) *peerGroupDialer {
+	return &peerGroupDialer{
+		lggr: lggr,
+
+		peerGroupFactory: peerGroupFactory,
+		rmnHomeReader:    rmnHomeReader,
+
+		bootstrapLocators:  bootstrapLocators,
+		oraclePeerIDs:      oraclePeerIDs,
+		commitConfigDigest: commitConfigDigest,
+
+		activePeerGroups: []rmn.PeerGroup{},
+
+		syncInterval: time.Minute, // todo: make it configurable
```

Comment:
A minute seems too long as a default, maybe default to Ethereum block time which is 12 seconds right now.